utils/Cockpit.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (C) 2024 Savoir-faire Linux Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.firefox.options import Options
+
+from utils.config import *
+
+class Cockpit(object):
+
+    def setup_cockpit_access(self):
+        options = Options()
+        options.accept_insecure_certs = True
+
+        self.driver = webdriver.Firefox(options=options)
+        self.driver.get(COCKPIT_URL)
+
+        element = self.driver.find_element(By.ID, "login-user-input")
+        element.clear()
+        element.send_keys(COCKPIT_USERNAME)
+
+        element = self.driver.find_element(By.ID, "login-password-input")
+        element.clear()
+        element.send_keys(COCKPIT_PASSWORD)
+
+        element = self.driver.find_element(By.ID, "login-button")
+        element.click()
+
+        try:
+            WebDriverWait(self.driver, 3).until(
+                EC.presence_of_element_located((By.ID, "toolbar"))
+            )
+            return self.driver
+        except:
+            self.driver.close()
+            print("connection to Cockpit failed")
+
+    def enable_administrative_access(self):
+        elements = self.driver.find_elements(By.CSS_SELECTOR, "button.ct-locked.pf-m-link.pf-c-button")
+        if elements:
```

Comment:
should we throw an error if the element is not found ?

Reply:
I don't think that is useful. If the element is not found, find_elements will return an empty list. In this particular case, there is nothing to do because this means administrative access are already enabled.